pep_sphinx_extensions/tests/test_pep_lint.py

```diff
@@ -0,0 +1,53 @@
+import importlib.util
+import sys
+from pathlib import Path
+
+# Import "pep-lint.py" as "pep_lint"
+PEP_LINT_PATH = Path(__file__).resolve().parent.parent.parent.joinpath("pep-lint.py")
+spec = importlib.util.spec_from_file_location("pep_lint", PEP_LINT_PATH)
+sys.modules["pep_lint"] = pep_lint = importlib.util.module_from_spec(spec)
+spec.loader.exec_module(pep_lint)
+
+
+def test_header_pattern_capitalisation():
+    pattern = pep_lint.HEADER_PATTERN
+
+    assert pattern.match("Header:")[1] == "Header"
+    assert pattern.match("header:")[1] == "header"
+    assert pattern.match("hEADER:")[1] == "hEADER"
+    assert pattern.match("hEaDeR:")[1] == "hEaDeR"
+
+
+def test_header_pattern_trailing_spaces():
+    pattern = pep_lint.HEADER_PATTERN
+
+    assert pattern.match("Header:")[1] == "Header"
+    assert pattern.match("Header: ")[1] == "Header"
+    assert pattern.match("Header:  ")[1] == "Header"
+
+
+def test_header_pattern_trailing_content():
+    pattern = pep_lint.HEADER_PATTERN
+
+    assert pattern.match("Header: Text")[1] == "Header"
+    assert pattern.match("Header: 123")[1] == "Header"
+    assert pattern.match("Header: !")[1] == "Header"
+    assert pattern.match("Header:Text") is None
+    assert pattern.match("Header:123") is None
+    assert pattern.match("Header:!") is None
+
+
+def test_header_pattern_colon_position():
+    pattern = pep_lint.HEADER_PATTERN
+
+    assert pattern.match("Header") is None
+    assert pattern.match("Header : ") is None
+    assert pattern.match("Header :") is None
+
+
+def test_header_pattern_separators():
+    pattern = pep_lint.HEADER_PATTERN
+
+    assert pattern.match("Hyphenated-Header:")[1] == "Hyphenated-Header"
+    assert pattern.match("Underscored_Header:") is None
+    assert pattern.match("Spaced Header:") is None
```

Comment:
These are a perfect fit for parameterised tests!

Separates test data from logic and easy to extend:

```suggestion
@pytest.mark.parametrize(
    "test_input, expected",
    [
        ("Header:", "Header"),
        ("header:", "header"),
        ("hEADER:", "hEADER"),
        ("hEaDeR:", "hEaDeR"),
        ("Header: ", "Header"),
        ("Header:  ", "Header"),
        ("Header: Text", "Header"),
        ("Header: 123", "Header"),
        ("Header: !", "Header"),
        ("Hyphenated-Header:", "Hyphenated-Header"),
    ],
)
def test_header_pattern(test_input, expected):
    pattern = pep_lint.HEADER_PATTERN

    assert pattern.match(test_input)[1] == expected


@pytest.mark.parametrize(
    "test_input",
    [
        "Header:Text",
        "Header:123",
        "Header:!",
        "Header",
        "Header : ",
        "Header :",
        "Underscored_Header:",
        "Spaced Header:",
    ],
)
def test_header_pattern_no_match(test_input):
    pattern = pep_lint.HEADER_PATTERN

    assert pattern.match(test_input) is None
```